crates/ibc/src/applications/transfer/amount.rs

```diff
@@ -6,10 +6,39 @@ use derive_more::{Display, From, Into};
 use super::error::TokenTransferError;
 use primitive_types::U256;
 
+#[cfg(feature = "schema")]
+use crate::alloc::{borrow::ToOwned, string::String};
+
 /// A type for representing token transfer amounts.
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
 #[derive(Copy, Clone, Debug, Eq, PartialEq, PartialOrd, Ord, Display, From, Into)]
-pub struct Amount(U256);
+pub struct Amount(
+    #[cfg_attr(feature = "schema", schemars(with = "String"))]
+    #[serde(serialize_with = "crate::serializers::serde_string::serialize")]
+    #[serde(deserialize_with = "deserialize")]
+    U256,
+);
+
+#[cfg(feature = "serde")]
+fn deserialize<'de, D>(deserializer: D) -> Result<U256, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    use serde::Deserialize;
+    U256::from_dec_str(<&str>::deserialize(deserializer)?).map_err(serde::de::Error::custom)
+}
```

Comment:
Can you place this under the [`seriliazers.rs`](https://github.com/cosmos/ibc-rs/blob/main/crates/ibc/src/serializers.rs#L0-L1) module? Also, `#[cfg(feature = "serde")]` not needed as having it derived on `seriliazers`

Reply:
i am not sure about it. any use for U256 from dec string rather than amount?

Reply:
until it public, not use for this https://github.com/cosmos/ibc-rs/pull/791#issuecomment-1658313262